components/Workshop/WorkshopListingInterest.tsx

```diff
@@ -0,0 +1,136 @@
+import { BsEmojiHeartEyes } from "react-icons/bs";
+import { GiConfirmed } from "react-icons/gi";
+import WorkshopListingInterestModal from "@components/Workshop/WorkshopListingInterestModal";
+
+interface IProps {
+  workshop: Workshop;
+  user: User | null;
+  isUserInterested: boolean;
+  numberOfInterestedUsers: number;
+}
+
+export default function WorkshopListingDescription({
+  workshop,
+  user,
+  isUserInterested,
+  numberOfInterestedUsers,
+}: IProps) {
+  const isUserHost = () => user?.id == workshop.user_id;
+  const toast = useToast();
+  const router = useRouter();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const interestedUsersCountText: string =
+    numberOfInterestedUsers == 1
+      ? `1 user expressed interest.`
+      : `${numberOfInterestedUsers} users expressed interest.`;
+
+  async function expressInterest(): Promise<void> {
+    try {
+      if (!user) {
+        router.push("/login");
+        return;
+      }
+      if (workshop.id) {
+        const interest: WorkshopInterest = {
+          workshop_id: workshop.id,
+          user_id: user.id,
+        };
+        const success = await clientData.expressInterestInWorkshop(interest);
+
+        if (success) {
+          toast({
+            title: "You have expressed interest",
+            description:
+              "Thank you for your interest in this workshop. The host will be notified.",
+            status: "success",
+            duration: 4000,
+            isClosable: true,
+          });
+          onClose();
+          router.refresh();
+        }
+      }
+    } catch (error) {
+      toast({
+        title: "Error",
+        description: "Problem expressing interest in this workshop.",
+        status: "error",
+        duration: 4000,
+        isClosable: true,
+      });
+    }
+  }
+
+  return (
+    <Box p="6" rounded="md" px={{ base: "6", md: "16" }}>
+      {!isUserHost() &&
+        (isUserInterested ? (
+          <Flex
+            justifyContent="space-between"
+            alignItems="center"
+            flexDirection="column"
+          >
+            <Button
+              rounded="full"
+              colorScheme="green"
+              variant="outline"
+              rightIcon={<GiConfirmed />}
+              size={{ base: "sm", sm: "md" }}
+              m="3"
+              isDisabled
+            >
+              I'm interested
+            </Button>
+          </Flex>
+        ) : (
+          <Flex
+            justifyContent="space-between"
+            alignItems="center"
+            flexDirection="column"
+          >
+            <Button
+              rounded="full"
+              colorScheme="blue"
+              variant="outline"
+              onClick={onOpen}
+              rightIcon={<BsEmojiHeartEyes />}
+              size={{ base: "sm", sm: "md" }}
+              m="3"
+            >
+              Show interest
+            </Button>
+            <WorkshopListingInterestModal
```

Comment:
I would replace this with `ConfirmActionDialog`. We already have a standard way of doing a modal so we should be reusing that. The modal should have a `Confirm` and optionally `Cancel` action (because to cancel you can also click on the X or to the side. You can find more examples of `ConfirmActionDialog` being used throughout the code.